grpc/grpc-go
# xds/internal/resolver: final bit of test cleanup

Summary of changes:
- Move the remaining xds resolver tests to `resolver_test` package
  - Remove previously used helpers defined here since they are no longer required
  - Modify existing tests to use new helpers defined in previous test cleanup PRs
  -  Split `TestXDSResolverHTTPFilters` into two
    - One tests config selector failure cases
    - The other tests HTTP filters
      - This is simplified to use a slice of `paths` instead of using a channel
- Add a helper to verify error received from resolver
- Create an internal package within `xds/internal/resolver` and move things to be overridden in tests to that package

RELEASE NOTES: none
